test/jdk/java/lang/module/exeNullCallerGetResource/exeNullCallerGetResource.c

```diff
@@ -0,0 +1,188 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "jni.h"
+#undef NDEBUG
+#include "assert.h"
+#include "string.h"
+
+static jclass class_InputStream = NULL;
+static jmethodID mid_InputStream_close = NULL;
+
+// in.close();
+void closeInputStream(JNIEnv *env, jobject in) {
+    (*env)->CallObjectMethod(env, in, mid_InputStream_close);
+    if ((*env)->ExceptionOccurred(env) != NULL) {
+        printf("ERROR: Exception was thrown calling InputStream::close.\n");
+        (*env)->ExceptionDescribe(env);
+        exit(-1);
+    }
+}
+
+/*
+ * The java test running this native test creates a test module named 'n'
+ * which opens the package 'open'.  It has a text file resource named
+ * 'test.txt' in the open package.  It also has a class called
+ * open.OpenResources.  One should be able to get the resource through
+ * either the Class or the Module with getResourceAsStream.
+ *
+ * Class c = open.OpenResources.fetchClass();
+ * InputStream in1 = c.getResourceAsStream("test.txt");
+ * Module n = c.getModule();
+ * InputStream in2 = n.getResourceAsStream("open/test.txt");
+ *
+ * The test also checks that closed resources are not available and
+ * don't throw any exceptions.  The test module contains a class
+ * called closed.ClosedResources and a file 'test.txt' in the package
+ * 'closed'.
+ *
+ * Class closed = closed.ClosedResources.fetchClass();
+ * assert(closed.getResourceAsStream("test.txt") == null);
+ * assert(n.getResourceAsStream("closed/test.txt") == null);
+ *
+ */
+int main(int argc, char** args) {
+    JavaVM *jvm;
+    JNIEnv *env;
+    JavaVMInitArgs vm_args;
+    JavaVMOption options[4];
+    jint rc;
+
+    options[0].optionString = "--module-path=mods";
+    options[1].optionString = "--add-modules=n";
+
+    vm_args.version = JNI_VERSION_9;
+    vm_args.nOptions = 2;
+    vm_args.options = options;
+
+    if ((rc = JNI_CreateJavaVM(&jvm, (void**)&env, &vm_args)) != JNI_OK) {
+        printf("ERROR: cannot create VM.\n");
+        exit(-1);
+    }
+
+    // initialize for stream close
+    class_InputStream = (*env)->FindClass(env, "java/io/InputStream");
+    assert(class_InputStream != NULL);
+    mid_InputStream_close = (*env)->GetMethodID(env, class_InputStream, "close", "()V" );
+    assert(mid_InputStream_close != NULL);
+
+    // Class c = OpenResources.fetchClass();
+    jclass class_OpenResources = (*env)->FindClass(env, "open/OpenResources");
+    assert(class_OpenResources != NULL);
+    jmethodID mid_OpenResources_fetchClass = (*env)->GetStaticMethodID(env, class_OpenResources, "fetchClass", "()Ljava/lang/Class;" );
```

Comment:
It seems that invoking `fetchClass` isn't necessary since you can simply use `class_OpenResources`.  Similarly for `class_ClosedResources`